src/components/MiningPositionList/index.js

```diff
@@ -0,0 +1,263 @@
+    cursor: pointer;
+    opacity: 0.6;
+  }
+
+  text-align: end;
+  user-select: none;
+`
+
+const DataText = styled(Flex)`
+  align-items: center;
+  text-align: center;
+  color: ${({ theme }) => theme.text1};
+  & > * {
+    font-size: 1em;
+  }
+
+  @media screen and (max-width: 600px) {
+    font-size: 13px;
+  }
+`
+
+const SORT_FIELD = {
+  VALUE: 'VALUE',
+  UNISWAP_RETURN: 'UNISWAP_RETURN'
+}
+
+function MiningPositionList({ miningPositions }) {
+  // const below500 = useMedia('(max-width: 500px)')
+  const below740 = useMedia('(max-width: 740px)')
+
+  // pagination
+  const [page, setPage] = useState(1)
+  const [maxPage, setMaxPage] = useState(1)
+  const ITEMS_PER_PAGE = 10
+
+  // sorting
+  const [sortDirection, setSortDirection] = useState(true)
+  const [sortedColumn, setSortedColumn] = useState(SORT_FIELD.VALUE)
+
+  useEffect(() => {
+    setMaxPage(1) // edit this to do modular
+    setPage(1)
+  }, [miningPositions])
+
+  useEffect(() => {
+    if (miningPositions) {
+      let extraPages = 1
+      if (miningPositions.length % ITEMS_PER_PAGE === 0) {
+        extraPages = 0
+      }
+      setMaxPage(Math.floor(miningPositions.length / ITEMS_PER_PAGE) + extraPages || 1)
+    }
+  }, [miningPositions])
+
+  const ListItem = ({ miningPosition, index }) => {
+    const pairPercentage = miningPosition.balance / 1e18 / miningPosition.pairData.totalSupply
+    const valueUSD = miningPosition.pairData.reserveUSD
+    const valueFirstPair = miningPosition.pairData.reserve0
+    const valueSecondPair = miningPosition.pairData.reserve1
+    const firstPairName = miningPosition.miningPool.pair.token0
+    const secondPairName = miningPosition.miningPool.pair.token1
+    const pairAddress = miningPosition.miningPool.pair.id
+    const firstPairAddress = miningPosition.pairData.token0.id
+    const secondPairAddress = miningPosition.pairData.token1.id
+
+    return (
+      <DashGrid style={{ opacity: pairPercentage > 0 ? 1 : 0.6 }} focus={true}>
+        {!below740 && <DataText area="number">{index}</DataText>}
+        <DataText area="name" justifyContent="flex-start" alignItems="flex-start">
+          <AutoColumn gap="8px" justify="flex-start" align="flex-start">
+            <DoubleTokenLogo size={16} a0={firstPairAddress} a1={secondPairAddress} margin={!below740} />
+          </AutoColumn>
+          <AutoColumn gap="8px" justify="flex-start" style={{ marginLeft: '20px' }}>
+            <CustomLink to={'/pair/' + pairAddress}>
+              <TYPE.main style={{ whiteSpace: 'nowrap' }} to={'/pair/'}>
+                <FormattedName text={firstPairName + '-' + secondPairName} maxCharacters={below740 ? 10 : 18} />
+              </TYPE.main>
+            </CustomLink>
+
+            <RowFixed gap="8px" justify="flex-start">
+              <Link external href={getMiningPoolLink(firstPairAddress)} style={{ marginRight: '.5rem' }}>
+                <ButtonLight style={{ padding: '4px 6px', borderRadius: '4px' }}>Stake</ButtonLight>
```

Comment:
Maybe should be "Stake More"

Reply:
I went ahead and changed that. But something about `Stake More` does not fit with the rest of the CTAs on the page IMO. I see the reasoning for adding "more" though. I will be thinking of something more appropriate. Ideally a one word term to be more consistent. For now we can go with this.

Reply:
Mhm, agreed its not ideal. I think theres a world where we could redesign this section entirely (for another day).